### ISSUE TITLE
Inaccurate Vertebrae Segmentations

### ISSUE THREAD

**Issue by Bacon-A** [wasserth/TotalSegmentator#268]
I am working on a vertebrae segmentation project and have been going through the publicly available dataset. I combined all of the vertebrae masks into one and viewed them all in ITK-SNAP to check for accuracy. Some didn't load (see below), so they haven't been checked yet. 
I separated my findings into two categories "Major Mask Issue" and "Minor Mask Issue".
In short, Major Errors I deem too incorrect to be trained on, while minor are kept in the training.
The carrots "^" indicate less serious, so from worst to best in order is Major, Major^, Major^^, minor, minor^, minor^^.
This was done once and not rechecked, so there may be some mistakes. The seriousness of the error is subjective, of course, and I also bias towards marking as a major issue if I don't think the image will assist in training the model (e.g. if only a small portion of vertebrae exists in the image, and it is not almost completely correct).

Overall the dataset is incredible. Thank you!

I have attached an example image of each.
Major Mask Issue (s0016)
<img width="529" alt="Major_mask_issue" src="https://github.com/wasserth/TotalSegmentator/assets/112503027/cf1e7c42-d600-4a3b-af81-bb9ca01cbf65">
Minor Mask Issue (s0070)
<img width="594" alt="image" src="https://github.com/wasserth/TotalSegmentator/assets/112503027/1ce28f2b-6cc0-480e-bf21-3c862a6fb857">

#### 0 corresponds to s0000. 1 corresponds to s0001, and so on.

Failed to Load (ITK-SNAP) - Non-Orthogonal Direction Matrix
0, 1, 2, 4, 13, 25, 43, 48, 61, 62,
66, 67, 79, 81, 90, 107, 112, 114, 117, 130,
146, 149, 154, 169, 189, 193, 208, 209, 226, 235,
236, 242, 254, 278, 282, 295, 299, 301, 311, 326, 
341, 352, 353, 361, 386, 417, 436, 457, 462, 504,
514, 526, 527, 552, 553, 557, 559, 561, 563, 577,
585, 603, 611, 614, 944, 949, 976, 978, 982, 986,
1000, 1005, 1009, 1011, 1023, 1035, 1038, 1040, 1046, 1050,
1083, 1103, 1133, 1141, 1147, 1170, 1174, 1205, 1234, 1235,
1237, 1272, 1276, 1295, 1309, 1344, 1389, 1406, 1407, 1409, 
1417, 1419, 1428, 

Major Mask Issue
11, 16, 28, 49, 59, 77, 84, 89, 95, 96, 
98, 128, 135, 145, 158^, 161, 171, 181, 184^, 196,
206, 210^, 219, 230^, 234, 238, 239, 240, 257, 259^^,
283, 296, 298, 335, 345, 350, 357, 358, 362, 366,
368^, 369, 377, 378^, 379^, 385, 389, 395^, 399, 400^,
408, 410^^, 414^^, 419, 428, 442, 446, 452^, 453, 454,
456, 460^^, 461^^, 474^^, 481^, 492^, 510^, 511^, 513,
515, 520, 521^^, 522, 525, 529, 531^, 532^^, 536, 546^^,
555, 564, 567, 575^^, 580, 585^^, 615^, 616, 618, 619,
627^, 632, 641, 644, 649^, 654^^, 656, 658^^, 664^^, 670,
671^, 674^^, 676, 679, 686^, 687, 688^, 691^^, 692^^, 693,
699^^, 700, 710^^, 714, 718, 723, 726^^, 727^, 728, 731,
732, 733, 740, 741, 747^, 748^, 749, 753, 754, 763^^
765, 766, 767^, 776^, 777, 781, 785, 788, 789^, 792,
793^, 796^, 800, 805, 806, 812^, 813, 815, 819, 820^^,
826, 840, 843, 852, 853, 859^, 861, 867, 870, 877^^,
878^, 879, 880^, 889, 890, 891, 892, 899, 903, 908^,
909^, 910, 918^^, 922^, 928^, 931^, 935, 936, 938, 940,
941, 953^, 956^^, 958, 962, 963, 965^, 968, 970^, 980^^,
984^^, 987, 993^, 995^, 996^, 997, 1006^, 1013, 1014, 1017,
1018, 1020, 1022, 1024, 1025, 1029^, 1031, 1036, 1039, 1041,
1042, 1043^^, 1045^, 1047, 1048, 1051, 1055^, 1063, 1065, 1072,
1075^, 1088, 1089, 1090, 1094^, 1097^^, 1102, 1105^, 1109, 1110^^,
1112^^, 1120, 1121, 1123^^, 1124^, 1128^^, 1132, 1143^, 1144^, 1148^,
1150^, 1155, 1164^, 1165^, 1167, 1169^, 1173, 1178, 1179^^, 1184,
1189^, 1201^, 1203^, 1206^^, 1209, 1212, 1216, 1220, 1221, 1222^, 
1224, 1226, 1233^, 1238, 1241, 1244^, 1246, 1248, 1249^, 1250, 
1251, 1258^, 1259, 1260^, 1264, 1267^, 1268^, 1270, 1275^, 1277^,
1280^, 1281, 1289^^, 1290, 1297^, 1304^, 1307, 1311^, 1312, 1314^,
1315, 1317^, 1319, 1325, 1326^, 1329, 1330, 1336, 1342^, 1346,
1350, 1357, 1361, 1364^, 1365, 1366, 1367^^, 1369^, 1370, 1371^, 
1372, 1375, 1379, 1387, 1390, 1395, 1397, 1399, 1405, 1408, 
1410, 1414, 1416, 1423, 

Minor Mask Issue
9, 29, 68^, 70, 71, 76, 83, 86, 99, 133, 136, 140, 167^, 
192, 223, 271, 281, 303, 305, 315, 216, 331, 344, 382, 396
407^^, 411, 422, 437, 439, 465^, 488^, 491, 493, 508, 517,
519, 535, 543, 544, 549, 551, 568, 572, 573, 591,
597, 613+, 622, 624, 629, 636, 642, 653, 655, 659,
660, 661, 669, 680, 684, 695, 698, 703^, 709^^, 716,
717, 720^, 743, 744, 771^^, 782, 790, 797^, 802^^, 807,
808, 816, 822, 824, 829, 831, 834, 847, 857, 863, 866,
874, 885, 896, 923^, 927, 933^, 934, 939, 946, 950,
952, 954, 957^, 972, 974, 991^^, 994, 1001, 1010, 1016,
1026, 1033, 1044, 1049, 1060^^, 1062^^, 1066^, 1071, 1077, 1082,
1101, 1111, 1113, 1114, 1115, 1119^^, 1127, 1138^^, 1140, 1149,
1154, 1159, 1162^^, 1168^^, 1171^, 1175^^, 1192, 1199, 1208^, 1210,
1223, 1230, 1231, 1240^, 1245^^, 1247, 1269, 1274, 1283^, 1285,
1287, 1294^^, 1301^, 1305^, 1323, 1327^, 1340^, 1341, 1343^, 1345,
1348, 1349, 1353, 1354, 1362^, 1363^^, 1382, 1388^, 1391, 1404, 
1412, 1415, 1418^,

**Comment by wasserth** [comment]
Thanks for your analysis. The vertebrae segmentations in the dataset your indeed not perfect. Getting them completely right is a lot of work. It turns out that even if the training dataset contains some errors, the resulting model is producing really good vertebrae segmentations (but still not perfect). This is good enough for many applications.
In the future we will try to further improve the segmentation quality in the training dataset which will contain improving the vertebrae groundtruth segmentations.
